src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -209,37 +209,8 @@ public async Task CreateLanguageSourceFilesAsync(GenerationLanguage language, Co
         private static readonly string requestBuilderSuffix = "RequestBuilder";
         private static readonly string voidType = "void";
         private static readonly string coreInterfaceType = "IHttpCore";
-        private CodeClass AddApiClientClass(CodeNamespace targetNS) {
-            var codeClass = targetNS.AddClass(new CodeClass(targetNS) { 
-                Name = config.ClientClassName,
-                ClassKind = CodeClassKind.RequestBuilder,
-                Description = "The main entry point of the SDK, exposes the configuration and the fluent API."
-            }).First();
-            var constructor = codeClass.AddMethod(new CodeMethod(codeClass) {
-                IsAsync = false,
-                Name = "constructor",
-                Access = AccessModifier.Public,
-                Description = "Instantiates a new Api client and sets the default values.",
-                IsStatic = false,
-                MethodKind = CodeMethodKind.ClientConstructor,
-                SerializerModules = config.Serializers,
-                DeserializerModules = config.Deserializers,
-            }).First();
-            constructor.ReturnType = new CodeType(constructor) { Name = voidType, IsExternal = true };
-            var httpCoreParameter = new CodeParameter(constructor) {
-                Name = "httpCore",
-                Description = "The http core service to use to execute the requests.",
-                Optional = false,
-                ParameterKind = CodeParameterKind.HttpCore,
-            };
-            httpCoreParameter.Type = new CodeType(httpCoreParameter) {
-                Name = coreInterfaceType,
-                IsExternal = true,
-            };
-            constructor.AddParameter(httpCoreParameter);
-            return codeClass;
-        }
-
+        private static readonly string httpCoreParameterName = "httpCore";
```

Comment:
those kind of corrections are handled at the refiner level. That's because I want to contain any language specific work to the refiners and writers to avoid a spread (and the maintenance burden that comes with it). see https://github.com/microsoft/kiota/blob/2d53c709fc52c642c855a5f8c821d52b8d337b6f/src/Kiota.Builder/Refiners/JavaRefiner.cs#L102